frontend/src/components/grantaccess/GrantApplicationAdmin.vue

```diff
@@ -0,0 +1,182 @@
+    applications.value = (
+        await AdminMgmtApiService.applicationsApi.getApplications()
+    ).data;
+});
+
+/* ------------------ User information method ------------------------- */
+const userIdChange = (userId: string) => {
+    formData.value.userId = userId;
+};
+
+const verifyUserIdPassed = ref(false);
+const setVerifyUserIdPassed = (verifiedResult: boolean) => {
+    verifyUserIdPassed.value = verifiedResult;
+};
+/* ---------------------- Form method ---------------------------------- */
+const cancelForm = () => {
+    formData.value = defaultFormData;
+    router.push('/dashboard');
+};
+
+const toRequestPayload = (formData: any) => {
+    const request = {
+        user_name: formData.userId,
+        application_id: formData.application.application_id,
+        user_type_code: UserType.I,
+    } as FamAppAdminCreateRequest;
+    return request;
+};
+
+const handleSubmit = async () => {
+    const data = toRequestPayload(formData.value);
+    await AdminMgmtApiService.applicationAdminApi
+        .createApplicationAdmin(data)
+        .then(() => {
+            setNotificationMsg(
+                Severity.success,
+                `Admin privilege has been added to ${formData.value.userId.toUpperCase()} for application ${
+                    formData.value.application.application_name
+                }`
+            );
+        })
+        .catch((error) => {
+            if (error.response?.status === 409) {
+                setNotificationMsg(
+                    Severity.warning,
+                    error.response.data.detail
+                );
+            } else if (
+                error.response.data.detail.code === 'self_grant_prohibited'
+            ) {
+                setNotificationMsg(
+                    Severity.error,
+                    'Granting admin privilege to self is not allowed.'
+                );
+            } else {
+                setNotificationMsg(
+                    Severity.success,
+                    `An error has occured. ${error.response.data.detail.description}`
+                );
+            }
+        })
+        .finally(() => {
+            router.push('/dashboard');
+        });
+};
+
+const onApplicationSelected = async (e: any) => {
+    formData.value.application = e.value;
+};
+</script>
+<template>
+    <PageTitle
+        title="Add application admin"
+        subtitle="Add a new application admin. All fields are mandatory unless noted"
+    />
+    <VeeForm
+        ref="form"
+        v-slot="{ errors, meta }"
+        :validation-schema="formValidationSchema"
+        as="div"
+    >
+        <div class="page-body">
+            <form id="grantAccessForm" class="form-container">
+                <StepContainer
+                    title="User information"
+                    subtitle="Enter the user information to add a new admin to [Application name]"
+                >
+                    <UserNameInput
+                        domain="I"
```

Comment:
let's use the `import { UserType } from 'fam-app-acsctl-api';` and `UserType.I`

Reply:
Thanks, I'll change it.